project5/3700kvstore

```diff
@@ -93,50 +144,56 @@ class Server:
-                str(self.state), str(state)))
-
-        if state == State.LEADER:
-            logging.info('Voted for by {}'.format(str(list(self.votes))))
-            # Re-initialize any necessary fields
-            self.state = State.LEADER
-            self.leader = self.id
-            self.prev_heartbeat = current_time()
-            for replica in self.replicas:
-                # For each server, index of next log entry to send to that server
-                # (initialized to leader last log index + 1)
-                self.next_idx[replica] = self.commit_idx + 1
-                # For each server, index of highest log entry known to be 
-                # replicated on server (initialized to 0, increases monotonically)
-                self.match_idx[replica] = 0
-
-            self.broadcast_heartbeat()
-            
-        elif state == State.FOLLOWER:
-            self.state = State.FOLLOWER
-            if msg:
-                self.leader = msg['leader']
-                self.current_term = msg['term']
-            self.votes = set()
-            self.voted_for = None
-
-        elif state == State.CANDIDATE:
-            self.state = State.CANDIDATE
-            self.leader = 'FFFF' # ID of the assumed leader
-            self.reset_timeouts()
-            if term is None:
-                self.current_term += 1
+        prev_state = self.state
+        prev_term = self.current_term
+        with self.accumulate_logs() as bypass_log:
+            if state == State.LEADER:
+                self.debug('Voted for by {}'.format(str(list(self.votes))))
+                # Re-initialize any necessary fields
+                self.state = State.LEADER
+                self.leader = self.id
+                self.prev_heartbeat = current_time()
+                for replica in self.replicas:
+                    # For each server, index of next log entry to send to that server
+                    # (initialized to leader last log index + 1)
+                    self.next_idx[replica] = self.commit_idx + 1
+                    # For each server, index of highest log entry known to be 
+                    # replicated on server (initialized to 0, increases monotonically)
+                    self.match_idx[replica] = 0
+    
+                self.broadcast_heartbeat()
+                
+            elif state == State.FOLLOWER:
+                if self.failed:
+                    bypass_log.info('Failed {}'.format(self.failed))
+                self.state = State.FOLLOWER
+                if msg:
+                    self.leader = msg['leader']
+                    self.current_term = msg['term']
+                self.votes = set()
+                self.failed = set()
+                self.voted_for = None
+    
+            elif state == State.CANDIDATE:
+                self.state = State.CANDIDATE
+                self.leader = 'FFFF' # ID of the assumed leader
+                self.reset_timeouts()
+                if term is None:
```

Comment:
The paper just says to increment `self.current_term`. When would `term` ever be `not None`?